src/mcts/node.cc

```diff
@@ -160,9 +162,58 @@ float Edge::GetP() const {
   return ret;
 }
 
+void Edge::MakeTerminal(GameResult result) {
+  certainty_state_ |= kTerminalMask | kCertainMask | kUpperBound | kLowerBound;
+  certainty_state_ &= kGameResultClear;
+  if (result == GameResult::WHITE_WON) {
+    certainty_state_ |= kGameResultWin;
+  } else if (result == GameResult::BLACK_WON) {
+    certainty_state_ |= kGameResultLoss;
+  }
+}
+
+void Edge::MakeCertain(GameResult result, CertaintyTrigger trigger) {
+  certainty_state_ |= kCertainMask | kUpperBound | kLowerBound;
```

Comment:
Consider encapsulating result and trigger as a small composite as there are multiple places where both are updated together. Example implementation available over here [https://github.com/Videodr0me/lc0/pull/3](https://github.com/Videodr0me/lc0/pull/3).

Reply:
If its compatible with PR487, i can do this, but i am not totally convinced it makes the code simpler. But if somebody else weighs in and also finds that this would be nicer, i will change it. Anybody?

Reply:
Especially the EvalPosition func gets easier to read ensuring the certaintyResult is mentioned as a composite of gameresult and certaintyState, although I do understand that mileage on readability may vary. As @ddobbelaere was actively commenting already; could you please have a look at the PR linked above and weigh in?

Reply:
Done!